contracts/Response.sol

```diff
@@ -189,16 +228,18 @@ contract Response {
     qs.highestLowestSupporterIdx = sIdx;
     setQuestionState(qIdx, qs);
   }
-
+  // QUESTION: why does parameter address exist/not have a name?
   function receiveApproval(address from, uint256 value, address, bytes extraData) public {
+    // QUESTION: the sender of this transaction will be the AE token contract?
     require(address(token) == msg.sender);
     require(token.transferFrom(from, this, value));
+    // QUESTION: does this assign qIdx as the last 32 bits from the extraData variable?
     uint qIdx;
     assembly { qIdx := mload(add(extraData, 32)) }
```

Comment:
As I understand, `extraData` points to a cell with a length of bytes array, `add(extraData, 32)` points to the first element of this array.